### PR TITLE
Use decimal separators for gold and XP

### PR DESCRIPTION
**Pull request by vlolteanu**
With the introduction of the stash, players can hoard huge amounts of gold. The total amount of gold can become illegible in the absence of decimal separators.
Screenshot_20220617_140658
Screenshot_20220617_142352

### CODE REVIEW

**Comment by AJenbo** [comment]
I like it, but i think locale is useless on most none Unix systems. Also @glebm removed it from usage recently so might be good to have some input from him as well.

**Comment by glebm** [comment]
`std::locale("")` is the user's locale, it does not necessary match the game language, right?

**Comment by AJenbo** [comment]
I think we have something for this already in the xp-bar code, maybe that can be reused for this.

**Comment by glebm** [comment]
Yeah, `PrintWithSeparator` is good (can be improved further by splitting into groups of 4 instead of 3 in CJK)

**Comment by vlolteanu** [comment]
XP would be another good candidate, but it doesn't fit in the box if you pass 1M.
Screenshot_20220617_212129

**Comment by vlolteanu** [comment]
I've also left the gold withdrawal inputs untouched. It feels weird to have commas placed there for you and it can't go over 5K (inventory) / 200K (stash) / 400K (hellfire stash)

File: Translations/fr.po [hunk 1]
```diff
@@ -10976,7 +10976,7 @@ msgid ""
 " \n"
 "Perhaps I can tell you more if we speak again. Good luck."
 msgstr ""
-"Dieu merci, vous êtes de retour !\n"
+"Dieu merci, vous êtes de retour !\n"
```
Comment (AJenbo) [inline Translations/fr.po:10979]: Any idea what the change here is?

**Review by AJenbo — COMMENTED** [review]
Thanks for also updating all the translation files

File: Translations/fr.po (hunk 1 above)
Comment (vlolteanu) [reply Translations/fr.po:10979]: Hexdump says it used to be a non-breaking space, and it got replaced with a space. Come to think of it, should there be a space there at all?

**Comment by vlolteanu** [comment]
@AJenbo No problem.

File: Translations/fr.po (hunk 1 above)
Comment (AJenbo) [reply Translations/fr.po:10979]: I don't speak French, but maybe @Phalcon22 knows :)

**Comment by vlolteanu** [comment]
I decreased the spacing for XP when over 1B, so that it fits.

File: Translations/fr.po (hunk 1 above)
Comment (vlolteanu) [reply Translations/fr.po:10979]: Apparently it's correct. https://www.quora.com/Why-do-French-speakers-place-a-space-before-exclamation-points

Comment (Phalcon22Epita) [reply Translations/fr.po:10979]: I confirm it is correct.

Comment (Phalcon22) [reply Translations/fr.po:10979]: I confirm it is correct.

Comment (AJenbo) [reply Translations/fr.po:10979]: Ok, if you can revert that to be NBS then I think we can merge this :)

Comment (vlolteanu) [reply Translations/fr.po]: Done.

File: Source/control.cpp [hunk 2]
```diff
@@ -1057,10 +1058,10 @@ void DrawGoldSplit(const Surface &out, int amount)
 	const std::string description = fmt::format(
 	    fmt::runtime(ngettext(
 	        /* TRANSLATORS: {:d} is a number. Dialog is shown when splitting a stash of Gold.*/
```
Comment (ephphatha) [inline Source/control.cpp:1060]:
```suggestion
	        /* TRANSLATORS: {:s} is a number with separators. Dialog is shown when splitting a stash of Gold.*/
```

**Comment by FitzRoyX** [comment]
For some reason I don't like this. Maybe it's because the font is so small that commas and decimals look the same. Maybe it's the inconsistency of only applying it to big numbers, or the fact that the ui is space-starved and can't afford it. There's also creating the uncertainty of typing withdrawals with or without commas.

**Review by AJenbo — APPROVED** [review]

**Comment by AJenbo** [comment]
If it helps you can pretend that it's also applied to all the small numbers since there is no difference with numbers that never go above 999

**Pull request merged**